oxidation/libparsec/crates/core_fs/src/storage/chunk_storage.rs

```diff
@@ -0,0 +1,409 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use diesel::dsl::count_star;
+use diesel::{sql_query, ExpressionMethods, Insertable, QueryDsl, RunQueryDsl};
+use std::sync::Mutex;
+
+use parsec_api_crypto::SecretKey;
+use parsec_api_types::{ChunkID, DateTime, DEFAULT_BLOCK_SIZE};
+
+use super::local_database::{SqliteConn, SQLITE_MAX_VARIABLE_NUMBER};
+use crate::error::{FSError, FSResult};
+use crate::extensions::CoalesceTotalSize;
+use crate::schema::chunks;
+
+#[derive(Insertable, AsChangeset)]
+#[diesel(table_name = chunks)]
+struct NewChunk<'a> {
+    pub chunk_id: &'a [u8],
+    pub size: i32,
+    pub offline: bool,
+    pub accessed_on: Option<f32>,
```

Comment:
could we use DateTime here instead ?

If not, could we use `DateTime` here if `DateTime` would implement `From<f64>` ? 😃 

Right now this is not super useful given there is few place where we do the conversion by hand, but it may become handy later on ^^

Reply:
Can we improve later ?